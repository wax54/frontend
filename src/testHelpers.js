import { render, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import UserContext from './UserContext';

const validUserContext = {
    user: {
        username: "wax54",
        firstName: "Sam",
        lastName: "Crewe-Sullam",
        email: "wax654@gmail.c om",
        isAdmin: false,
        applications: [1, 2, 3]
    },
    isLoggedIn: true
}

function preRender(options={}) {
    const defaultOptions = {
        initialEntries: ['/'],
        context: {
            user: {},
            isLoggedIn: false
        }
    }
    const newOptions = {
        initialEntries: 
            [...(options.initialEntries || []), ...defaultOptions.initialEntries],
        context: 
            { ...defaultOptions.context, ...(options.context || {}) }
    }
    //console.log('options: ', options, 'newOptions', newOptions);
    return newOptions;
}

function renderWithRouter(Component, options) {
    options = preRender(options);
    return render(
        <MemoryRouter initialEntries={options.initialEntries}>
            <Component />
        </MemoryRouter>
    );
}

async function renderAndWaitFor(Component, options) {
    options = preRender(options);
    return await waitFor(() => render(
        <MemoryRouter initialEntries={options.initialEntries}>
            <Component />
        </MemoryRouter>
    ));
}


function renderWithUserContext(Component, options) {
    options = preRender(options);
    return render(
        <MemoryRouter initialEntries={options.initialEntries}>
            <UserContext.Provider value={options.context} >
                <Component />
            </UserContext.Provider>
        </MemoryRouter>
    );
}

async function renderWithUserContextAndWait(Component, options) {
    options = preRender(options);
    return await waitFor(() => render(
        <MemoryRouter initialEntries={options.initialEntries}>
            <UserContext.Provider value={options.context} >
                <Component />
            </UserContext.Provider>
        </MemoryRouter>
    ));
}



export { validUserContext, 
    renderWithRouter, 
    renderAndWaitFor, 
    renderWithUserContext, 
    renderWithUserContextAndWait }