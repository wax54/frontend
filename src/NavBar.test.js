import { validUserContext, renderWithUserContext } from './testHelpers';
import NavBar from './NavBar';

test("doesn't blow up", () => {
    renderWithUserContext(NavBar);
    renderWithUserContext(NavBar, { context: validUserContext });
});

test("matchesSnapshot", () => {
    const { asFragment } = renderWithUserContext(NavBar);
    expect(asFragment()).toMatchSnapshot();
});

test("matchesSnapshot with valid user", () => {
    const { asFragment } = renderWithUserContext(NavBar, { context: validUserContext });
    expect(asFragment()).toMatchSnapshot();
});