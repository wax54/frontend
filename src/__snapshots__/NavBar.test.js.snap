// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matchesSnapshot 1`] = `
<DocumentFragment>
  <div
    class="NavBar"
  >
    <a
      aria-current="page"
      class="active"
      href="/"
    >
      Home
    </a>
    <a
      href="/login"
    >
       login
    </a>
    <a
      href="/signup"
    >
      Signup
    </a>
  </div>
</DocumentFragment>
`;

exports[`matchesSnapshot with valid user 1`] = `
<DocumentFragment>
  <div
    class="NavBar"
  >
    <a
      aria-current="page"
      class="active"
      href="/"
    >
      Home
    </a>
    <a
      href="/companies"
    >
      companies
    </a>
    <a
      href="/jobs"
    >
      jobs
    </a>
    <a
      href="/profile"
    >
      Hello, wax54!
    </a>
    <a
      href="/"
    >
      logout
    </a>
  </div>
</DocumentFragment>
`;
